demos/coreMQTT_Agent/mqtt_agent_task.c

```diff
@@ -0,0 +1,1025 @@
+/*
+ * FreeRTOS V202104.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/*
+ * This demo creates multiple tasks, all of which use the MQTT agent API to
+ * communicate with an MQTT broker through the same MQTT connection.
+ *
+ * This file contains the initial task created after the TCP/IP stack connects
+ * to the network.  The task:
+ *
+ * 1) Connects to the MQTT broker.
+ * 2) Creates the other demo tasks, in accordance with the #defines set in
+ *    demo_config.h.  For example, if demo_config.h contains the following
+ *    setting:
+ *
+ *    #define democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE 3
+ *
+ *    then the initial task will create three instances of the task
+ *    implemented in simple_sub_pub_demo.c.  See the comments at the top
+ *    of that file for more information.
+ *
+ * 3) After creating the demo tasks the initial task could create the MQTT
```

Comment:
Is this paragraph valid about the initial task assuming the role of the agent task? This demo calls `xTaskCreate` on the `prvMqttAgentTask` function, thereby creating a separate task for the agent

Reply:
Not any more